### PR TITLE
Canny destination doc

### PR DESCRIPTION
**Pull request by ishachopra0401**
## What do these changes do?

Created new doc for Canny destination

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/destinations/streaming-destinations/canny.mdx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+<div class="successBlock">
+You can specify multiple <strong>Canny Events</strong> for one <strong>Event Name</strong> and vice versa.
+</div>
+
+## Identify
+
+You can use the <Link to="/event-spec/standard-events/identify/">`identify`</Link> call to find the `id` of a user:
+- If a user does not exist, it is created and its `id` is returned. 
+- If a user exists, it is updated with new data.
```
Comment (ItsSudip) [inline docs/destinations/streaming-destinations/canny.mdx:52]: We can create and update a user only through identify call.

File: docs/destinations/streaming-destinations/canny.mdx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+| `properties.imageURLs` | `imageURLs` | Optional | Array of strings |
+
+### Property mappings for create vote Event 
+
+The following table lists the mappings between RudderStack and Canny properties for **Create vote** event:
+
+| RudderStack property | Canny property | Presence | Data type | 
+|:---------------------|:-------------|:--------------|:--------------|
+| `properties.postId` | `postID` | Required | String |
+| Collect from `externalId` or <Link to="https://developers.canny.io/api-reference#retrieve_user">retrieve user</Link> API call | `voterID` | Required | String |
```
Comment (ItsSudip) [inline docs/destinations/streaming-destinations/canny.mdx:133]: The key name of externalId is `cannyUserId` we have to mention that somewhere.

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: docs/destinations/streaming-destinations/canny.mdx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+| RudderStack property | Canny property | Presence | Data type | Notes |
+|:---------------------|:-------------|:--------------|:--------------|:--------------|
+| `properties.boardID`/`properties.boardId`/`properties.board.id` | `boardID` | Required | String | -|
+| Collect from `externalId` or <Link to="https://developers.canny.io/api-reference#retrieve_user">retrieve user</Link> API call| `authorID` | Required | String |-|
+| `properties.details` | `details` | Required | String |-|
+| `properties.title` | `title` | Required | String |-|
+| `properties.byID`/`properties.byId`/`properties.by.id` | `byID` | Optional | String |-|
+| `properties.categoryID`/`properties.categoryId`/`properties.category.id` | `categoryID` |Optional | String |-|
+| `properties` (after removing `boardID`, `byID`, `categoryID`,`details`, `eta`, `etaPublic`, `title`, `imageURLs`)| `customFields` | Optional | Object | Ensure that you <Link to="#how-to-create-custom-fields-in-canny-dashboard">create the custom fields in Canny dashboard</Link> to send the event seuccessfully.|
```
Comment (ItsSudip) [inline docs/destinations/streaming-destinations/canny.mdx:119]: for `customFields` we are looking for `customFields` in properties so need to update the mapping.
```suggestion
| `properties.customFields` | `customFields` | Optional | Object | Ensure that you <Link to="#how-to-create-custom-fields-in-canny-dashboard">create the custom fields in Canny dashboard</Link> to send the event seuccessfully.|
```

**Review by ItsSudip — CHANGES_REQUESTED** [review]

**Review by ItsSudip — APPROVED** [review]

File: docs/destinations/streaming-destinations/canny.mdx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+- **API Key**: Enter your API key from the Canny dashboard.
+
+<div class="infoBlock">
+Refer to the <Link to="#faq">FAQ</Link> section below for more information on getting the API Key.
+</div>
+
+### Event settings
+
+- **Mapping to trigger Canny Events for the respective Event**: Enter the event name and pick the Canny event from the dropdown to be triggered when that event is called.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/canny.mdx:43]:
```suggestion
- **Mapping to trigger Canny Events for the respective Event**: Enter the event name and choose the Canny event from the dropdown to be triggered when that event is called.
```

File: docs/destinations/streaming-destinations/canny.mdx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+<div class="successBlock">
+You can specify multiple <strong>Canny Events</strong> for one <strong>Event Name</strong> and vice versa.
+</div>
+
+## Identify
+
+You can use the <Link to="/event-spec/standard-events/identify/">`identify`</Link> call to <Link to="https://developers.canny.io/api-reference#create_or_update_user">create or update a user</Link>.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/canny.mdx:51]:
```suggestion
You can use the <Link to="/event-spec/standard-events/identify/">`identify`</Link> call to <Link to="https://developers.canny.io/api-reference#create_or_update_user">create or update a user</Link> in Canny.
```

File: docs/destinations/streaming-destinations/canny.mdx [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+The following table lists the mappings between RudderStack and Canny properties:
+
+| RudderStack property | Canny property | Presence | Data type | 
+|:---------------------|:-------------|:--------------|:--------------|
+| `userId`/`traits.userId`/`traits.id`/`context.traits.userId`/`context.traits.id` | `userId` | Required | String |
+| `context.traits.name`/`traits.name` | `name` |Required | String |
+| `context.traits.email`/`traits.email`/`properties.email` | `email` | Optional | String |
+| `context.traits`/`traits` (after removing `email` and `name`) | `customFields` |Optional | Object |
+| `originalTimestamp`/`timestamp` | `created` | Optional | Date |
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/canny.mdx:83]: Should this be in the ISO 8601 timestamp format? @ItsSudip @rohithkaza cc @ishachopra0401 we should specify this accordingly.

File: docs/destinations/streaming-destinations/canny.mdx [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+    details: "Require new feature",
+    title: "New Feature",
+    eta: "10.08.2022",
+    priority: "High"
+})
+```
+
+### Property mappings for Create post event 
+
+The following table lists the mappings between RudderStack and Canny properties for **Create post** event:
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/canny.mdx:110]:
```suggestion
The following table lists the mappings between RudderStack and Canny properties for the **Create post** event:
```

File: docs/destinations/streaming-destinations/canny.mdx [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+
+| RudderStack property | Canny property | Presence | Data type | Notes |
+|:---------------------|:-------------|:--------------|:--------------|:--------------|
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/canny.mdx:113]: we can drop the **Notes** column here.

File: docs/destinations/streaming-destinations/canny.mdx [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+| `properties.boardID`/`properties.boardId`/`properties.board.id` | `boardID` | Required | String | -|
+| Collect from `externalId` or <Link to="https://developers.canny.io/api-reference#retrieve_user">retrieve user</Link> API call| `authorID` | Required | String |-|
+| `properties.details` | `details` | Required | String |-|
+| `properties.title` | `title` | Required | String |-|
+| `properties.byID`/`properties.byId`/`properties.by.id` | `byID` | Optional | String |-|
+| `properties.categoryID`/`properties.categoryId`/`properties.category.id` | `categoryID` |Optional | String |-|
+| `properties.customFields` | `customFields` | Optional | Object | Ensure that you <Link to="#how-to-create-custom-fields-in-canny-dashboard">create the custom fields in Canny dashboard</Link> to send the event seuccessfully.|
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/canny.mdx:120]:
```suggestion
| `properties.customFields` | `customFields` | Optional | Object | Ensure that you <Link to="#how-to-create-custom-fields-in-canny-dashboard">create the custom fields in Canny dashboard</Link> to send the event successfully.|
```

File: docs/destinations/streaming-destinations/canny.mdx [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+| `properties.eta` | `eta` | Optional | String | In MM/YYYY format. Required if `etaPublic` is present. |
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/canny.mdx:121]:
```suggestion
| `properties.eta` | `eta` | Required if `etaPublic` is present. | String | Should be in MM/YYYY format.  |
```

File: docs/destinations/streaming-destinations/canny.mdx [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+| `properties.etaPublic` | `etaPublic` | Optional | Boolean | Required if `eta` is present. |
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/canny.mdx:122]:
```suggestion
| `properties.etaPublic` | `etaPublic` | Required if `eta` is present.  | Boolean | - |
```

File: docs/destinations/streaming-destinations/canny.mdx [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+| `properties.imageURLs` | `imageURLs` | Optional | Array of strings | -|
+
+### Property mappings for Create vote event 
+
+The following table lists the mappings between RudderStack and Canny properties for **Create vote** event:
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/canny.mdx:127]:
```suggestion
The following table lists the mappings between RudderStack and Canny properties for the **Create vote** event:
```

File: docs/destinations/streaming-destinations/canny.mdx [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+### Where can I find the Canny API Key?
+
+To find the Canny API Key:
+
+1. Log into your [Canny account](https://canny.io/login).
+2. Go to **Settings** > **API & Webhooks** under your profile, as shown: 
+
+<img src="../../assets/event-stream-destinations/canny-api-key.png" alt="Canny dashboard"/>
+
+### How to create custom fields in Canny Dashboard?
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/canny.mdx:149]:
```suggestion
### How to create custom fields in the Canny dashboard?
```

File: docs/destinations/streaming-destinations/canny.mdx [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+
+To create custom fields in Canny Dashboard:
+
+1. Log into your [Canny account](https://canny.io/login).
+2. Go to **Settings** > **Boards** under your profile.
+3. Select the desired **Board** from the dropdown and select **Create Post Form**.
+4. Click on **Create new Post Fields to add them to a board** in the **Fields** section.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/canny.mdx:156]:
```suggestion
4. In the **Fields** section, click on **Create new Post Fields to add them to a board**.
```

File: docs/destinations/streaming-destinations/canny.mdx (hunk 6 above)
Comment (ItsSudip) [reply docs/destinations/streaming-destinations/canny.mdx:83]: Yes, it should be in the ISO 8601 timestamp format.

Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/canny.mdx:83]:
```suggestion
| `originalTimestamp`/`timestamp` | `created` | Optional | ISO 8601 Timestamp |
```

**Review by ameypv-rudder — APPROVED** [review]

File: docs/destinations/streaming-destinations/canny.mdx [hunk 15]
```diff
@@ -0,0 +1,159 @@
...
+| `context.traits.companies`/`traits.companies` | `companies` | Optional | Array of object (in company objects `id` and `name` are the only required fields) |
+
+## Track
+
+You can create the following events using the <Link to="/event-spec/standard-events/track/">`track`</Link> call:
+
+- <Link to="https://developers.canny.io/api-reference#create_post">Create post</Link>
+- <Link to="https://developers.canny.io/api-reference#create_vote">Create vote</Link>
+ 
+Also, you can retrieve the user data using `userId` or `email` and use that information to post or vote.
```
Comment (rohithkaza) [inline docs/destinations/streaming-destinations/canny.mdx:94]:
```suggestion
We <Link to="/event-spec/standard-events/track/">`rerieve the user`</Link> data using `userId` or `email` and use that information to create post or vote.
```

File: docs/destinations/streaming-destinations/canny.mdx [hunk 16]
```diff
@@ -0,0 +1,159 @@
...
+### How to create custom fields in the Canny dashboard?
+
+To create custom fields in Canny Dashboard:
+
+1. Log into your [Canny account](https://canny.io/login).
+2. Go to **Settings** > **Boards** under your profile.
+3. Select the desired **Board** from the dropdown and select **Create Post Form**.
+4. In the **Fields** section, click on **Create new Post Fields to add them to a board**.
+5. Click on **Create New Field**, as shown:
+
```
Comment (ItsSudip) [inline docs/destinations/streaming-destinations/canny.mdx:158]: We need to do another step:
Your Company/Dashboard -> Settings-> Boards -> [Select a Board From Dropdown] -> Create Post Form -> Add Field -> [Select a Post Field From Dropdown] -> Enter details -> Save

**Pull request merged**